### ISSUE TITLE
Private async functions not working

### ISSUE THREAD

**Issue by fxOne** [babel/babel#9354]
## Bug Report

**Current Behavior**
```
Module parse failed: Can not use keyword 'await' outside an async function (240:2)
You may need an appropriate loader to handle this file type.
| 
| var _asyncFn2 = function _asyncFn2() {
>   await Promise.resolve();
| };
```

**Input Code**
```js
class test {
  async #asyncFn() {
    await Promise.resolve();
  }
}
```

**Expected behavior/code**
I would expect that private async functions works like non private async function.

**Babel Configuration (.babelrc, package.json, cli command)**

```js
{
  'env': {
    'development': {
      plugins:['babel-plugin-transform-builtin-extend', {
        'globals': ['Error', 'Array']
      }],
        ['@babel/plugin-syntax-async-generators'],
        ['@babel/plugin-proposal-unicode-property-regex'],
        ['@babel/plugin-proposal-object-rest-spread'],
        ['@babel/plugin-proposal-decorators', {decoratorsBeforeExport: true}],
        ['@babel/plugin-proposal-class-properties'],
        ['@babel/plugin-transform-runtime'],
        ['@babel/plugin-syntax-dynamic-import'],
        ['@babel/plugin-proposal-private-methods'],
        ['lodash'] ,
        'presets': [['@babel/preset-env', {
        'targets': {
          'browsers': ['last 2 Chrome versions']
        },
        'modules': false
      }]],
        'compact': false,
        'comments': true,
        'sourceMaps': true
    }
  }
}
```

**Environment**
- Babel version(s): @babel/core 7.2.2, @babel/plugin-proposal-private-methods 7.2.3
- Node/npm version: Node 10/npm 5
- OS: Ubuntu 18.04
- Monorepo: no
- How you are using Babel: WEbpack

**Comment by littledan** [comment]
cc @tim-mc

**Comment by dnalborczyk** [comment]
fwiw, same is true for `private generator` and `private async generator` methods:

```javascript
class A {
  async #foo() {}
  *#bar() {}
  async *#foobar() {}
}
```

**Comment by nicolo-ribaudo** [comment]
Fixed by #9423
